src/Sparrow/Json/BlittableWriter.cs

```diff
@@ -286,31 +314,28 @@ public static void CleanPropertyArrayOffset()
             _intBuffer = null;
         }
 
+#if NET6_0_OR_GREATER
+        [SkipLocalsInit]
+#endif
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private int WritePropertyNames(int rootOffset)
+        private unsafe int WritePropertyNames(int rootOffset)
         {
             var cachedProperties = CachedProperties;
             int propertiesDiscovered = cachedProperties.PropertiesDiscovered;
 
             // Write the property names and register their positions
-            if (_propertyArrayOffset == null || _propertyArrayOffset.Length < propertiesDiscovered)
-            {
-                _propertyArrayOffset = new int[Bits.NextAllocationSize(propertiesDiscovered)];
-            }
+            int* propertyArrayOffset = stackalloc int[propertiesDiscovered];
```

Comment:
I believe we always tried to allocate aligned buffers. Can you shed more light on the motivation for dropping `Bits.NextAllocationSize` here?